packages/elements/src/datetime-field/index.ts

```diff
@@ -0,0 +1,708 @@
+              break;
+            // no default
+          }
+          this.setValueAndNotify(newValue);
+          this.syncInputValue();
+          this.selectPart(selectedPartIndex, this.formatToParts(newValue));
+          this.resetError();
+          event.preventDefault();
+        }
+        break;
+      case 'ArrowRight':
+      case 'ArrowLeft':
+        const nextPartIdx = getNextSelectedPartIndex(selection, parts, this.inputValue, key === 'ArrowLeft' ? Direction.Down : Direction.Up);
+        this.selectPartFrame.schedule(() => {
+          this.selectPart(nextPartIdx, parts);
+        });
+        break;
+      // no default
+    }
+  }
+
+  /**
+   * Check if value is changed and fire event
+   * @returns {void}
+   */
+  // eslint-disable-next-line @typescript-eslint/no-unused-vars
+  protected override onPossibleValueChange (event: InputEvent): void {
+    // Nobody likes to see a red border
+    this.resetError();
+    this.selectPartFrame.cancel(); // ensure no pending selection
+    this.partLabel = '';
+
+    const inputValue = this.inputElement?.value || '';
+    const parsedValue = this.toValue(inputValue);
+    this.setValueAndNotify(parsedValue);
+  }
+
+  /**
+   * Handles key down input event
+   * @param event Key down event object
+   * @returns {void}
+   */
+  protected onInputKeyDown (event: KeyboardEvent): void {
+    if (this.readonly || this.disabled || event.defaultPrevented) {
+      return;
+    }
+
+    // Ignore special keys
+    if (event.shiftKey || event.ctrlKey || event.altKey || event.metaKey) {
+      return;
+    }
+
+    switch (event.key) {
+      case 'Up': // IE11
+      case 'ArrowUp':
+        this.onNavigation('ArrowUp', event);
+        break;
+      case 'Down':
+      case 'ArrowDown':
+        this.onNavigation('ArrowDown', event);
+        break;
+      case 'Left':
+      case 'ArrowLeft':
+        this.onNavigation('ArrowLeft', event);
+        break;
+      case 'Right':
+      case 'ArrowRight':
+        this.onNavigation('ArrowRight', event);
+        break;
+      case 'Esc':
+      case 'Escape':
+        this.onNavigation('Escape', event);
+        break;
+      default:
+        return;
+    }
+  }
+
+  /**
+   * Decorate `<input>` element with common properties.
+   * Override placeholder
+   * @returns template map
+   */
+  protected override get decorateInputMap (): TemplateMap {
+    return {
+      ...super.decorateInputMap,
+      placeholder: this.focused && !this.inputValue ? this.toInputValue(this.startDate) : this.placeholder || undefined,
+      '@keydown': this.onInputKeyDown
+    };
+  }
+
+  /**
+   * Used to announce part and value statuses
+   */
+  protected get screenReaderTemplate (): TemplateResult | null {
+    if (!this.focused) {
+      return null;
+    }
+    return html`
+      ${this.partLabel && this.value ? html`<div
+        aria-label="${this.t(`PICK_${this.partLabel.toUpperCase()}`)}"
```

Comment:
My editor says 

`You are binding a non-primitive type 'DirectiveResult<{ prototype: any; new(partInfo: ChildPartInfo | AttributePartInfo | ElementPartInfo) => AsyncTranslateDirective; }>'. This could result in binding the string "[object Object]"`

I know we shouldn't cast `DirectiveResult` to `string` but can we handle this somehow ?

Reply:
Interesting why does it say this. The syntax is valid and we use it in many places, perhaps something somewhere has changed....

Reply:
FYI, warning is part of `lit-plugin` in VSCode.

Reply:
maybe a plugin has a bug

Reply:
I guess we can leave it as it is.